src/prompt-storage.ts

```diff
@@ -15,3 +19,33 @@ export const checkPromptMessageCompleted = (
   const id = Cookies.get(`bucket-prompt-${userId}`);
   return id === promptId;
 };
+
+export const rememberAuthToken = (
+  userId: string,
+  channel: string,
+  token: string,
+  expiresAt: Date,
+) => {
+  Cookies.set(`bucket-token-${userId}`, `${channel}:${token}`, {
+    expires: expiresAt,
+    sameSite: "strict",
+    secure: true,
+  });
+};
+
+export const getAuthToken = (userId: string) => {
+  const val = Cookies.get(`bucket-token-${userId}`);
+  if (!val) {
+    return undefined;
+  }
+
+  const [channel, token] = val.split(":");
+  if (!channel?.length || !token?.length) {
+    throw new Error(`Invalid token: ${val}`);
```

Comment:
Do we want to throw as part of this interface? We run the risk of not catching it somewhere and getting into a non-working state in the client.

Also, If we have a known non-working cookie payload, maybe we should clear the cookie at this point

Reply:
You're right, I wanted to be through since the cookie would be set by us and such an incorrect value would be an error on our side. But, I think it makes sense to simply ignore it and return `undefined` which would be equivalent to - `no cookie`.